### PR TITLE
fix warnings and type inconsistencies

### PR DESCRIPTION
**Pull request by nicolasdanet**
In order to build the sigmund~.c file as CPP in "Spaghettis" i changed few things:

    1. Remove unused (or unreachable) values.
    2. Explicit some casts.
    3. And moreover change t_sample to t_float for buffering.

Note that in my version t_sample are single precision whereas t_float are double precision.

That changes could be incorporated in #522

### CODE REVIEW

File: extra/sigmund~/sigmund~.c [hunk 1]
```diff
@@ -838,7 +838,7 @@ typedef struct _sigmund
     int x_npts;         /* number of points in analysis window */
     int x_npeak;        /* number of peaks to find */
     int x_loud;         /* debug level */
-    t_sample *x_inbuf;  /* input buffer */
+    t_float *x_inbuf;  /* input buffer */
```
Comment (umlaeute) [inline extra/sigmund~/sigmund~.c]: this doesn't look right.
i think it should be done the other way around (more `t_sample` rather than less)

Comment (nicolasdanet) [reply extra/sigmund~/sigmund~.c]: I'm not agree but it is up to you (feel free to close this PR).

**Comment by umlaeute** [comment]
why did you close this?

**Comment by nicolasdanet** [comment]
C++ support seems already done in #597 ; The other modifications was required only for me and i don't use that file anymore.

**Pull request closed without merging**